File: go/binary_search/find_last_test.go
package binary_search

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestFindLast_RandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData()

		expected := td.expectedLast
		actual := FindLastInt(td.array, td.val)
		if actual != expected {
			assert.FailNowf(t, "",
				"FindLastInt(%v, %v):\n  Actual: %v\nExpected: %v",
				td.array, td.val, actual, expected,
			)
		}
	}
}

File: python/binary_search/find_last_test.py

from bs_random_test_data import RandomTestData
from find_last import find_last_int


def test_find_first_random_cases():
	for _ in range(10_000):
		td = RandomTestData()

		want = td.expected_last
		got = find_last_int(td.array, td.val)
		assert got == want
